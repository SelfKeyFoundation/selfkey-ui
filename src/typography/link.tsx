import * as React from 'react';
// @ts-ignore
import injectSheet, { StyleSheet, StyledComponentProps } from 'react-jss';

const styles: StyleSheet = {
	a: {
		color: '#fff',
		fontFamily: 'ProximaNovaRegular, arial, sans-serif',
		fontSize: '16px',
		fontWeight: 400,
		textDecoration: 'underline',
	},
};

export type LinkProps = {
	className?: string;
	href?: string;
	onClick?: ((event: React.MouseEvent<HTMLElement>) => void);
};

export const A = injectSheet(styles)<LinkProps>(({ classes, children, className, href, onClick }) => (
	<a className={`${classes.base} ${classes.a} ${className}`} href={href} onClick={onClick}>
		{children}
	</a>
));
