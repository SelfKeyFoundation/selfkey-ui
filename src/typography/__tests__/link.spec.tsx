import * as React from 'react';
import { create } from 'react-test-renderer';
import { A } from '../link';


describe('Link', () => {
  it('should show Link', () => {
    const component = create(
      <A href="http://test" onClick={(event)=>{}}>TEST</A>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
