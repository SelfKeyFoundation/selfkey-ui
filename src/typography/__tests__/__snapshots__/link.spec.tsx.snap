// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link should show Link 1`] = `
<a
  className="undefined Component-a-0-1-1 undefined"
  href="http://test"
  onClick={[Function]}
>
  TEST
</a>
`;
